lib/snaplist_bloc.dart

```diff
@@ -143,26 +145,41 @@ class SnapListBloc {
 
   _calculateScrollProgress(double currentPosition) {
     final distance = (_startPosition - currentPosition).abs();
-    return ((distance * 100) /
-            sizeProvider(_centerItemPosition, _createBuilderData()).width)
-        .clamp(0.0, 100.0);
+    double sizeFactor = isVertical ?
+      sizeProvider(_centerItemPosition, _createBuilderData()).height :
+      sizeProvider(_centerItemPosition, _createBuilderData()).width;
+    return ((distance * 100) / sizeFactor).clamp(0.0, 100.0);
   }
 
   double _calculateTargetOffset() {
     double result = 0.0;
 
     for (var i = 1; i <= _nextItemPosition; ++i) {
-      double cardWidth = sizeProvider(
-          i - 1,
-          BuilderData(
-            _centerItemPosition,
-            _nextItemPosition,
-            100.0,
-          )).width;
+      if (isVertical) {
```

Comment:
The result of the `sizeProvider` call may be stored in a `final cardSize`, and then the `width` or `height` might be read from there. I'll shrink this method twice in size.